addons/im_livechat/models/im_livechat_channel.py

```diff
@@ -560,6 +560,18 @@ def get_livechat_info(self, username=None):
     def _is_livechat_available(self):
         return self.chatbot_script_count or len(self.available_operator_ids) > 0
 
+    def _get_non_persisted_channel_info(self, /, *, channel_id, operator_partner, chatbot_data, **kwargs):
+        return {
+            "fetchChannelInfoState": "fetched",
+            "id": channel_id,
+            "isLoaded": True,
+            "livechat_operator_id": Store.One(
+                operator_partner, self.env["discuss.channel"]._store_livechat_operator_id_fields(),
+            ),
+            "scrollUnread": False,
+            "channel_type": "livechat",
+            "chatbot": chatbot_data,
+        }
```

Comment:
This should not be moved, especially on this model if self is never used.

Just do the same pattern as extra_operator_lookup_params where the "extra" method stays in the controller and the base code is not moved.